AEROGEAR-9623 edit conflict data sync (#733)
aerogear/mobile-docs: modules/ROOT/pages/_partials/data-sync/js-client-conflict.adoc

Developers can either use the default conflict resolution implementations, or implement their own conflict resolution implementations using the conflict resolution mechanism.

By default, when no changes are made on the same fields, the implementation attempts to resend the modified payload back to the server. 
When changes on the server and on the client affect the same fields, one of the specified conflict resolution strategies can be used. 
The default strategy applies client changes on top of the server side data. 
Developers can modify strategies to suit their needs.

[#working-with-conflicts-client]
== Implementing conflict resolution on the client

To enable conflict resolution, the server side resolvers must be configured to perform conflict detection. 
Detection can rely on different implementations and return the conflict error back to the client. 
See <<#conflict-resolution-{context}, Server Side Conflict Detection>> for more information.

> You must provide mutations with the context, that is the `returnType` parameter to resolve conflicts.

Reviewer: ```suggestion
If you want to use conflict resolution, you must provide the client with the `returnType` of the object being operated on. This can be done in one of two ways.
```

Reviewer: The wording here might not be 100% but the `context` is something they could provide in other scenarios (or for their own use cases) so we should say `returnType` is something else you can provide the `context` with.

Author: updated, wdyt?

})
----

> If using Apollo's mutate function, provide the the `returnType` paramter as follows:

Reviewer: ```suggestion
If using Apollo's mutate function, provide the `returnType` parameter as follows:
```

})
----

> The client automatically resolves the conflicts based on the current strategy and notify listeners as required.

Reviewer: ```suggestion
The client automatically resolves the conflicts based on the current strategy and notifies listeners as required.
```
